### PR TITLE
ft: S3C-4551 tiny version IDs

### PR DESCRIPTION
**Pull request by vrancurel**
The tiny mode is NOT selected by default when enabling versioning. It is for now configurable for deployments that need it.

### CODE REVIEW

**Comment by vrancurel** [comment]
There is probably much better/simpler/more elegant approaches. Please suggest.

File: lib/Config.js [hunk 1]
```diff
@@ -1284,11 +1287,32 @@ class Config extends EventEmitter {
     }
 }
 
+// get the version ID encoding type
+function getVersionIdEncType(verCfg) {
+    if (verCfg === undefined) {
+        throw new Error('Invalid version configuration');
```
Comment (jonathan-gramain) [inline lib/Config.js]: it looks dangerous to me to throw here, e.g. at https://github.com/scality/cloudserver/pull/3660/files#diff-d9d6c7d27a37666321fe0d79f3d76b7349a03856a0933381651431fb4e5cff62R341 are we sure that there is always a defined versioning configuration?

Comment (vrancurel) [reply lib/Config.js]: I realize that `decode()` is sometimes called without a versioningConfiguration. Problematic.

Comment (vrancurel) [reply lib/Config.js]: So if you look carefully I somehow managed to call `getEncType()` only when needed and it does not assert any more.

File: lib/Config.js [hunk 2]
```diff
@@ -1284,11 +1287,27 @@ class Config extends EventEmitter {
     }
 }
 
+// get the version ID encoding type
+function getVersionIdEncType() {
+    let encType;
+    if (process.env.VERSION_ID_ENC_TYPE !== undefined) {
```
Comment (anurag4DSB) [inline lib/Config.js]: are the possible values 0 or 1?
If yes, will note for documentation purposes.

**Review by anurag4DSB — COMMENTED** [review]
What should we do about older objects?
will we create a migration tool?

File: lib/Config.js (hunk 2 above)
Comment (vrancurel) [reply lib/Config.js]: Yes, but I think we need set ENC_TYPE_HEX by default. I am just trying to pass the tests with BASE62 then I will set HEX by default (for now).

**Comment by vrancurel** [comment]
> What should we do about older objects?
> will we create a migration tool?

This is the same binary format. Only the presentation changes.

But we would need a flag per bucket to state if we shall use the old or new representation implicitly or explicitly.
This state can be managed by a bucket property "BucketFeature" that we can use for other breaking changes.

**Review by saitharunthodupunuru — APPROVED** [review]
LGTM

**Review by miniscruff — APPROVED** [review]

**Review by saitharunthodupunuru — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Comment by rahulreddy** [comment]
Bypassing as Azure flakiness work is coming

**Pull request merged**